### PR TITLE
CCP-29 feat/front & API (Inscription) : add inscription page

### PR DESCRIPTION
**Pull request by hippolyte94**
This is the first version of the Inscription page with : 
- Little verification for password (can be upgrade later)

Little problem with the implentation of the DropBox - she will be there in the first Pull Request

### CODE REVIEW

File: front/src/View/Pages/Inscription.tsx [hunk 1]
```diff
@@ -0,0 +1,197 @@
+import React, {ChangeEvent, FormEvent, ReactNode} from 'react';
+import Request from '../../API/Request';
+import '../../css/Inscription.css';
+import DropBox from '../General/DropBox';
+
+class Inscription extends React.Component{
```
Comment (djodjo89) [inline front/src/View/Pages/Inscription.tsx]: use states

File: front/src/View/Pages/Inscription.tsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        this.nom = event.target.value;
+    }
+
+    public handlePseudoChange (event: ChangeEvent<HTMLInputElement>) : void {
+        this.pseudo = event.target.value;
+    }
+
+    public passwordVerification () : boolean {
+
+        if (this.password.length > 8){
```
Comment (djodjo89) [inline front/src/View/Pages/Inscription.tsx]: maybe use a sleekier ternary expression ?

File: front/src/View/Pages/Inscription.tsx [hunk 3]
```diff
@@ -0,0 +1,197 @@
+import React, {ChangeEvent, FormEvent, ReactNode} from 'react';
+import Request from '../../API/Request';
+import '../../css/Inscription.css';
+import DropBox from '../General/DropBox';
+
+class Inscription extends React.Component{
+    private email: string;
+    private password: string;
+    private passwordConfirmation: string;
+    private prenom: string;
```
Comment (djodjo89) [inline front/src/View/Pages/Inscription.tsx]: write in english

File: front/src/View/Pages/Inscription.tsx [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    public submitForm(event: FormEvent<HTMLFormElement>): void {
+        event.preventDefault();
+
+        if (this.checkForm()){
+            new Request(
+                '/connection/register',
+                this.updateConnectStatus,
+                'POST',
+                {
+                    prenom : this.prenom,
```
Comment (djodjo89) [inline front/src/View/Pages/Inscription.tsx]: respect api's specifications

File: front/src/View/Pages/Inscription.tsx [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+                    nom : this.nom,
+                    pseudo : this.pseudo,
+                    email: this.email,
+                    password: this.password
+                },
+            );
+        }
+        else{
```
Comment (djodjo89) [inline front/src/View/Pages/Inscription.tsx]: use swalAlert

File: front/src/View/Pages/Inscription.tsx [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+            alert("erreur")
+        }
+
+        this.setState({formWasSubmitted: true});
```
Comment (djodjo89) [inline front/src/View/Pages/Inscription.tsx]: use it as a callback

File: front/src/View/Pages/Inscription.tsx [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+                        <br/>
+                        Déja Inscrit ? <a id={"redirect-button"} onClick={this.redirectToConnectionPage}>Connecte toi ! </a>
+                    </div>
+                </div>
+            </section>
+        )
+    }
+}
+
+interface InscriptionInputProps {
```
Comment (djodjo89) [inline front/src/View/Pages/Inscription.tsx]: use more generic Input api

File: front/src/View/Pages/Inscription.tsx [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+                   onFocus={e => e.target.placeholder = ""}
+                   onBlur={e => e.target.placeholder = this.props.placeholder}
+                   onChange={this.props.onChange}
+            />
+        </div>
+
+    }
+}
+
+class ButtonInscription extends React.Component<{}, {}> {
```
Comment (djodjo89) [inline front/src/View/Pages/Inscription.tsx]: use more generic SubmitButton api

File: front/src/View/Pages/Inscription.tsx [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+    public render() {
+        return (
+            <button type="submit" className="btn btn-default btn-transparent mt-0 rounded-1 custom">Connexion</button>
```
Comment (djodjo89) [inline front/src/View/Pages/Inscription.tsx]: Should'nt it be "Créer mon compte" or something like this ?

File: front/src/View/Pages/Inscription.tsx [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+        this.email = event.target.value;
+    }
+
+    public handlePasswordChange(event: ChangeEvent<HTMLInputElement>): void {
+        this.password = event.target.value;
+    }
+
+    public handlePasswordConfirmationChange (event: ChangeEvent<HTMLInputElement>) : void {
+        this.passwordConfirmation = event.target.value;
+        this.passwordVerification();
```
Comment (djodjo89) [inline front/src/View/Pages/Inscription.tsx]: use a state and call password verification as a callback

File: front/src/View/Pages/Inscription.tsx (hunk 1 above)
Comment (hippolyte94) [reply front/src/View/Pages/Inscription.tsx]: but we don't hace state in connection page no ? for the input variable i mean

File: front/src/View/Pages/Inscription.tsx (hunk 3 above)
Comment (hippolyte94) [reply front/src/View/Pages/Inscription.tsx]: done

File: front/src/View/Pages/Inscription.tsx (hunk 5 above)
Comment (hippolyte94) [reply front/src/View/Pages/Inscription.tsx]: done

File: front/src/View/Pages/Inscription.tsx (hunk 9 above)
Comment (hippolyte94) [reply front/src/View/Pages/Inscription.tsx]: done !

**Pull request closed without merging**